### PR TITLE
Pin to sphinx=2.4.4 to avoid problem with C declaration

### PR DESCRIPTION
**Pull request by sklam**
This unbreaks master.

### CODE REVIEW

**Comment by esc** [comment]
I tried this locally with the correct sphinx version and can confirm that the warnings don't appear with it. Also, judging from public CI the version change was made in the correct place as the `doc` build now passes. 👍

File: buildscripts/incremental/setup_conda_environment.sh [hunk 1]
```diff
@@ -78,7 +78,7 @@ fi
 $CONDA_INSTALL -c numba llvmlite
 
 # Install dependencies for building the documentation
-if [ "$BUILD_DOC" == "yes" ]; then $CONDA_INSTALL sphinx sphinx_rtd_theme pygments numpydoc; fi
+if [ "$BUILD_DOC" == "yes" ]; then $CONDA_INSTALL sphinx=2.4.4 sphinx_rtd_theme pygments numpydoc; fi
```
Comment (stuartarchibald) [inline buildscripts/incremental/setup_conda_environment.sh:81]: This change needs making on windows `setup_conda_environment.cmd` too?

**Comment by stuartarchibald** [comment]
@esc I've updated this patch series with 347b6b5 for windows, please could you re-review? Thanks.

**Comment by stuartarchibald** [comment]
Abandon all that, forgot that the windows script isn't actually in use as it doesn't work on Azure. Going with original patch in 8586057, once re-tested will merge.

**Pull request merged**